File: module_bean/src/main/java/com/example/william/my/bean/data/LoginBean.java
package com.example.william.my.bean.data;

import com.google.gson.annotations.SerializedName;

public class LoginBean {

    @SerializedName("data")
    private LoginUserBean data;

    public LoginUserBean getUserData() {
        return data;
    }

    public void setUserData(LoginUserBean data) {
        this.data = data;
    }
}

File: modules/module_sample/src/main/java/com/example/william/my/module/sample/bean/LoginBean.kt
package com.example.william.my.module.sample.bean

import android.os.Parcelable
import androidx.annotation.Keep
import kotlinx.parcelize.Parcelize

@Keep
@Parcelize
class LoginBean(val data: User) : Parcelable {

    @Keep
    @Parcelize
    class User(val id: String, val nickname: String) : Parcelable
}